doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc

```diff
@@ -0,0 +1,93 @@
+
+=== Retrieve the current state of a representation
+
+The `FormDescription#getVariableManagerInitializer` API allows to initialize variables for a given form representation.
+
+We can leverage this form-only (for now) mechanism to retrieve the `IRepresentationEventProcessor` of the representation we want the state of:
+
+```java
+Function<VariableManager, VariableManager> variableManagerInitializer =
+  vm -> {
+      String editingContextId = vm.get(IEditingContext.EDITING_CONTEXT, IEditingContext.class).map(IEditingContext::getId).orElse("");
+
+      IEditingContextEventProcessor editingContextEventProcessor = this.editingContextEventProcessorRegistry.getEditingContextEventProcessors().stream()
+              .filter(processor -> processor.getEditingContextId().equals(editingContextId))
+              .findFirst()
+              .orElse(null);
+      IRepresentationEventProcessor representationEventProcessor = Optional.ofNullable(editingContextEventProcessor)
+              .flatMap(processor -> processor.getRepresentationEventProcessors().stream()
+                      .filter(IRepresentationEventProcessor.class::isInstance)
+                      .map(IRepresentationEventProcessor.class::cast)
+                      .filter(p -> p.getRepresentation().getId().equals(SOME_REPRESENTATION_ID_RETRIEVED_FROM_THE_FORM_SUBSCRIPTION_PARAMETERS))
+                      .findFirst()
+              )
+              .orElse(null);
+      vm.put("editingContextEventProcessor", editingContextEventProcessor);
+      vm.put("representationEventProcessor", representationEventProcessor);
+      vm.put("representation", representationEventProcessor.getRepresentation());
+  }
+```
+
+The `representation` variable could the be used in the form to initialize its content (e.g. populate a tree widget).
+
+=== Propagate changes on an existing representation
+
+The section above allows to create new representations that use the content of existing representations to intialize their content, but it may be necessary to apply mutations on the existing representation from actions in the new one (e.g. a form initialized with a diagram may have a button to change the visibility of elements on the diagram).
+In this case we want to make sure both representation stay synchronized.
+
+This can be done by leveraging the `IRepresentationEventProcessor#handle` method to perform an operation on another representation, but doing so requires to create a new ChangeDescription flux, which won't be known by the `EditingContextEventProcessor`.
+
+This means that the synchronization won't be complete: the operation will be executed, but the representation calling the operation won't be automatically updated. 
+
+Synchronization can be achieved done by manually calling `IRepresentationEventProcessor#refresh` with the local ChangeDescription flux on all the `IRepresentationEventProcessor` of the current `IEditingContextProcessor`:
```

Comment:
`can be achieved done by` -> `can be achieved by`

Reply:
done